interview-questions.md

```diff
@@ -8,36 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: We would generate a rails migration that adds a id to the cohort table. The name would be cohort_id. It would be on the student model. This actually happened to me twice on two rails projects we worked on. I created a new column and migrated it onto the 2nd table to reference the first table. On our credit card table each user had many credit cards but I forgot to add an ID so that I could add multiple credit cards to each user.
 
-Researched answer:
+Researched answer: To fix this, we need to add a foreign key to the students table. To add the key to the students table we need to creat a migration. Rails g migration add_cohort_id_to_students cohort:references would be the command we use to add a cohort id column to the students table as a foreign key referencing the cohorts table. We can then run rails db:migrate.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: Update, when you add an update action you need to have its own route. 
 
-Researched answer:
+Researched answer:In RESTful routing some routes require paramters to be passed in the request. These are used to identify specific resources. They must be passed for the request. Routes that alwyas require paramaters are the ones the perform actions on specifc resources. Example editing or deleting.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: Rails generate model, makes the table and the columns in the table.
+(thats probably the only one I can name off the top of my head)
 
-Researched answer:
+Researched answer: Rails Generate Resource, this will create everything generate model does but also creates a resource in the routes.rb file. You can also assign foerign keys. Rails destroy, this will go through every path generate used but destroy it instead. Rails generate controller, this will create a new controller file. 
```

Comment:
rails g resource actually creates everything a rails g model AND a rails g controller create.  Just fyi. :)